File: CAEVSYNC.Web/clientapp/src/models/EventTransformationStepModel.ts
import { PropertyType } from "./PropertyType";
import { EventTransformationType } from "./EventTransformationType";

export interface EventTransformationStepModel {
    id: string;
    propertyName: string;
    propertyType: PropertyType;
    transformationType: EventTransformationType;
    intReplacement: number | null;
    stringReplacement: string | null;
    boolReplacement: boolean | null;
    intFilter: number | null;
    stringFilter: string | null;
    boolFilter: boolean | null;
    fromDateTime: Date | null;
    toDateTime: Date | null;
    extraMinutesBefore: number | null;
    extraMinutesAfter: number | null;
}

File: CAEVSYNC.Common/Models/EventTransformationStepModel.cs
using CAEVSYNC.Common.Models.Enums;

namespace CAEVSYNC.Common.Models;

public class EventTransformationStepModel
{
    public string Id { get; set; }
    
    public string PropertyName { get; set; }
    
    public PropertyType PropertyType { get; set; }
    
    public EventTransformationType TransformationType { get; set; }
    
    public int? IntReplacement { get; set; }
    
    public string? StringReplacement { get; set; }
    
    public bool? BoolReplacement { get; set; }
    
    public int? IntFilter { get; set; }
    
    public string? StringFilter { get; set; }
    
    public bool? BoolFilter { get; set; }
    
    public DateTime? FromDateTime { get; set; }
    
    public DateTime? ToDateTime { get; set; }

    public int? ExtraMinutesBefore { get; set; }
    
    public int? ExtraMinutesAfter { get; set; }
}